ffi/src/handle.rs

```diff
@@ -367,8 +365,10 @@ mod private {
         type Raw = T;
 
         fn into_handle_ptr(val: Arc<T>) -> NonNull<T> {
-            let val = ManuallyDrop::new(val);
-            val.as_ref().into()
+            let ptr = Arc::into_raw(val);
+            // Note: casting ptr as `*mut T` is needed for NonNull, and actually Arc::into_raw
+            // _does_ create a mutable pointer (via `Arc::as_ptr`), so this is an 'okay' cast.
+            unsafe { NonNull::new_unchecked(ptr as *mut T) } // into_raw guarantees non-null
```

Comment:
Suggest to use pointer's [cast_mut](https://doc.rust-lang.org/std/primitive.pointer.html#method.cast_mut) method, because:
> This is a bit safer than `as` because it wouldn’t silently change the type if the code is refactored.

```rust
            // Note: casting ptr as `*mut T` is needed for NonNull, and actually Arc::into_raw
            // _does_ create a mutable pointer (via `Arc::as_ptr`), so this is an 'okay' cast.
            let ptr = Arc::into_raw(val).cast_mut();
            unsafe { NonNull::new_unchecked(ptr) } // into_raw guarantees non-null